packages/sign-client/src/controllers/engine.ts

```diff
@@ -572,6 +591,402 @@ export class Engine extends IEngine {
     return this.client.pendingRequest.getAll();
   };
 
+  // ---------- Auth ------------------------------------------------ //
+
+  public authenticate: IEngine["authenticate"] = async (params) => {
+    this.isInitialized();
+    this.isValidAuthenticate(params);
+
+    const { chains, statement = "", uri, domain, nonce, type, exp, nbf, methods = [] } = params;
+    // reassign resources to remove reference as the array is modified and might cause side effects
+    const resources = [...(params.resources || [])];
+
+    const { topic: pairingTopic, uri: connectionUri } = await this.client.core.pairing.create({
+      methods: ["wc_sessionAuthenticate"],
+    });
+
+    this.client.logger.info({
+      message: "Generated new pairing",
+      pairing: { topic: pairingTopic, uri: connectionUri },
+    });
+
+    const publicKey = await this.client.core.crypto.generateKeyPair();
+    const responseTopic = hashKey(publicKey);
+
+    await this.client.auth.authKeys.set(AUTH_PUBLIC_KEY_NAME, { responseTopic, publicKey });
+    await this.client.auth.pairingTopics.set(responseTopic, { topic: responseTopic, pairingTopic });
```

Comment:
Is there a reason to do this sequentially or can we use `Promise.all` ?
```suggestion
    await Promise.all([
      this.client.auth.authKeys.set(AUTH_PUBLIC_KEY_NAME, { responseTopic, publicKey }),
      this.client.auth.pairingTopics.set(responseTopic, { topic: responseTopic, pairingTopic })
    ]);
```